src/main/java/org/spongepowered/api/block/entity/BlockEntityArchetype.java

```diff
@@ -115,13 +130,62 @@ static Builder builder() {
 
         Builder blockEntity(BlockEntityType type);
 
+        /**
+         * Copies the incoming {@link BlockEntity} for it's current state
+         * of information, such as {@link BlockState}, {@link BlockEntityType},
+         * and all related {@link DataView contained data}, except the position
+         * and potentially any Sponge-added tracking information, such as owner
+         * or notifier.
+         *
+         * <p>Note that this overwrites data set from the following methods:
+         * <ul>
+         *     <li>{@link #from(Location)}</li>
+         *     <li>{@link #state(BlockState)}</li>
+         *     <li>{@link #blockEntity(BlockEntityType)}</li>
+         *     <li>{@link #blockEntityData(DataView)}</li>
+         *     <li>{@link #set(Value)}</li>
+         *     <li>{@link #set(Key, Object)}</li>
+         *     <li>{@link #set(DataManipulator)}</li>
+         * </ul></p>
+         * @param blockEntity The block entity to absorb all data
+         * @return This builder, for chaining
+         */
         Builder blockEntity(BlockEntity blockEntity);
 
+        /**
+         * Sets the {@link DataView} to be used to copy data onto the created
+         * {@link BlockEntity} when {@link BlockEntityArchetype#apply(Location)}
+         * is called.
+         *
+         * <p>The format used for this container follows the
+         * <a href="https://minecraft.gamepedia.com/Chunk_format#Block_entity_format">Mojang ChunkFormat for BlockEntities.</a>,
+         * and can be customized based on the origination of the {@link BlockEntity}.
+         * If the block entity originates from a content-adding mod, the format could
+         * vary based on it's implementation and may change at any time.
+         * </p>
+         *
+         * @param dataView The data view containing data pertinent to the
+         *     {@link BlockEntity}
+         * @return This builder, for chaining
+         */
         Builder blockEntityData(DataView dataView);
 
-        Builder set(DataManipulator manipulator);
-
-        <E, V extends Value<E>> Builder set(V value);
+        /**
+         * Sets the {@link DataManipulator}'s {@link Value}s onto this builder,
+         * to apply to the resulting {@link BlockEntity} when the built
+         * {@link BlockEntityArchetype} is used.
+         *
+         * @param manipulator The manipulator containing values
+         * @return This builder, for chaining
+         */
+        default Builder set(DataManipulator manipulator) {
```

Comment:
`set`/`add` should be used consistently across builders, e.g. `DataBuilder.Immutable` uses `add`